models/taskRequests.js

```diff
@@ -0,0 +1,142 @@
+const { TASK_REQUEST_STATUS } = require("../constants/taskRequests");
+const { TASK_STATUS } = require("../constants/tasks");
+const { userState } = require("../constants/userStatus");
+const firestore = require("../utils/firestore");
+const taskRequestsCollection = firestore.collection("taskRequests");
+const tasksModel = require("./tasks");
+const userModel = require("./users");
+const userStatusModel = require("./userStatus");
+
+/**
+ * Fetch all task requests
+ *
+ * @return {Object}
+ */
+const fetchTaskRequests = async () => {
+  try {
+    const taskRequests = await taskRequestsCollection.get();
+
+    const taskRequestsData = [];
+
+    taskRequests.forEach((taskRequest) => taskRequestsData.push({ id: taskRequest.id, ...taskRequest.data() }));
+
+    return taskRequestsData;
+  } catch (err) {
+    logger.error("error fetching tasks", err);
+    throw err;
+  }
+};
+
+/**
+ * Creates a task request
+ *
+ * @param taskId { string }: id of task request
+ * @return {Promise<{taskRequest: Object}>}
+ */
+const createTaskRequest = async (taskId, userId) => {
+  try {
+    const taskRequest = await taskRequestsCollection.doc(taskId).get();
+    const taskRequestData = taskRequest.data();
+
+    const { userExists, user } = await userModel.fetchUser({ userId });
+
+    if (!userExists) {
+      return { userDoesNotExists: true };
+    }
+
+    if (taskRequestData) {
+      const currentRequestors = taskRequestData.requestors;
+      if (currentRequestors.length >= 1) {
+        const requestorExists = !!currentRequestors.find((requestor) => requestor === user.id);
```

Comment:
Rename from `requestorExists` to `requestor`

Reply:
This is a boolean value since `!!` operator is being provided, so I thought `requestorExists` is more appropriate name than `requestor`.

Reply:
Oh, I missed this part.